.github/workflows/nextjs_bundle_analysis.yml

```diff
@@ -0,0 +1,117 @@
+              path: .next/cache
+              # change this if you prefer a more strict cache
+              key: ${{ runner.os }}-build-${{ env.cache-name }}
+
+         - name: Build next.js app
+           # change this if your site requires a custom build command
+           run: ./node_modules/.bin/next build
+           env:
+              ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_API_KEY }}
+              NEXT_PUBLIC_STRAPI_ORIGIN: https://main.govinor.com
+
+         # Here's the first place where next-bundle-analysis' own script is used
+         # This step pulls the raw bundle stats for the current bundle
+         - name: Analyze bundle
+           run: npx -p nextjs-bundle-analysis report
+
+         - name: Upload bundle
+           uses: actions/upload-artifact@v2
+           with:
+              name: bundle
+              path: .next/analyze/__bundle_analysis.json
+
+         - name: Download base branch bundle stats
+           uses: dawidd6/action-download-artifact@v2
+           if: success() && github.event.number
+           with:
+              workflow: nextjs_bundle_analysis.yml
+              branch: ${{ github.event.pull_request.base.ref }}
+              path: .next/analyze/base
+
+         # And here's the second place - this runs after we have both the current and
+         # base branch bundle stats, and will compare them to determine what changed.
+         # There are two configurable arguments that come from package.json:
+         #
+         # - budget: optional, set a budget (bytes) against which size changes are measured
+         #           it's set to 350kb here by default, as informed by the following piece:
+         #           https://infrequently.org/2021/03/the-performance-inequality-gap/
+         #
+         # - red-status-percentage: sets the percent size increase where you get a red
+         #                          status indicator, defaults to 20%
+         #
+         # Either of these arguments can be changed or removed by editing the `nextBundleAnalysis`
+         # entry in your package.json file.
+         - name: Compare with base branch bundle
+           if: success() && github.event.number
+           run: ls -laR .next/analyze/base && npx -p nextjs-bundle-analysis compare
+
+         - name: Get comment body
+           id: get-comment-body
+           if: success() && github.event.number
+           run: |
+              body=$(cat .next/analyze/__bundle_analysis_comment.txt)
+              body="${body//'%'/'%25'}"
+              body="${body//$'\n'/'%0A'}"
+              body="${body//$'\r'/'%0D'}"
+              echo ::set-output name=body::$body
+
+         - name: Find Comment
+           uses: peter-evans/find-comment@v2
+           if: success() && github.event.number
+           id: fc
+           with:
+              issue-number: ${{ github.event.number }}
+              body-includes: '<!-- __NEXTJS_BUNDLE -->'
+
+         - name: Create Comment
```

Comment:
The `peter-evans/create-or-update-comment` action allows for the `creation` and `updating` steps to be combined based on the output of `Find Comment`. [Link](https://github.com/peter-evans/create-or-update-comment#:~:text=If%20the%20find%2Dcomment%20action%20output%20comment%2Did%20returns%20an%20empty%20string%2C%20a%20new%20comment%20will%20be%20created.%20If%20it%20returns%20a%20value%2C%20the%20comment%20already%20exists%20and%20the%20content%20is%20replaced.)

We do this in `iFixit/ifixit` as well https://github.com/iFixit/ifixit/blob/master/.github/actions/issue-comment/action.yml#L23

> If the find-comment action output comment-id returns an empty string, a new comment will be created. If it returns a value, the comment already exists and the content is replaced.

```yaml
    - name: Create or Update comment
        uses: peter-evans/create-or-update-comment@v2
        if: success() && github.event.number
        with:
           issue-number: ${{ github.event.number }}
           comment-id: ${{ steps.fc.outputs.comment-id }}
           edit-mode: replace
           body: ${{ steps.get-comment-body.outputs.body }}
```

Reply:
This action version is also a bit outdated, we are currently using `v2` in `iFixit/ifixit`.

Reply:
FYI, the bulk of this workflow was generated by: https://github.com/hashicorp/nextjs-bundle-analysis

So, while updating it seems like a good idea, diverging from the generated file makes updates harder. Thoughts?

Reply:
I did see that most of this workflow was generated from the `npx -p nextjs-bundle-analysis generate` command. However, it was generated from their template file where this particular section was last updated more than a year ago.

<img width="1097" alt="Git Blame@2x" src="https://user-images.githubusercontent.com/22064420/208187938-462916d4-b617-4d3a-91a0-1f6a7274fc75.png">

The commenting part does not really have anything to do with the logic associated with how the GitHub action reports or compares the bundle size. Therefore, I think updating this section is fine.